src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,845 @@
+        pos = self._applyInversion(self.position.value)
+
+        for n, i in self._axis_map.items():
+            if axes is None or n in axes:
+                pos[n] = self.GetPosition(i) / self._axis_conv_factor[i]
+
+        pos = self._applyInversion(pos)
+        pos = self._applyOffset(pos)  # Appy the offset back for display
+
+        logging.debug("Updated position to %s", pos)
+
+        self.position._set_value(pos, force_write=True)
+
+    def _cancelCurrentMove(self, future):
+        """
+        Cancels the current move (both absolute or relative). Non-blocking.
+        future (Future): the future to stop. Unused, only one future must be
+         running at a time.
+        return (bool): True if it successfully cancelled (stopped) the move.
+        """
+        # The difficulty is to synchronise correctly when:
+        #  * the task is just starting (not finished requesting axes to move)
+        #  * the task is finishing (about to say that it finished successfully)
+        logging.debug("Cancelling current move")
+
+        future._must_stop.set()  # tell the thread taking care of the move it's over
+        with future._moving_lock:
+            if not future._was_stopped:
+                logging.debug("Cancelling failed")
+            return future._was_stopped
+
+    @isasync
+    def reference(self, axes):
+        model.Actuator.reference(self, axes)
+
+    def stop(self, axes=None):
+        self._executor.cancel()
+        # For safety, just force stop every axis
+        for an, aid in self._axis_map.items():
+            if axes is None or an in axes:
+                self.StopMotion(aid)
+        
+    def _createMoveFuture(self):
+        """
+        Return (CancellableFuture): a future that can be used to manage a move
+        """
+        f = CancellableFuture()
+        f._moving_lock = threading.Lock()  # taken while moving
+        f._must_stop = threading.Event()  # cancel of the current future requested
+        f._was_stopped = False  # if cancel was successful
+        f.task_canceller = self._cancelCurrentMove
+        return f
+
+
+class ESPSimulator(object):
+    """
+    Simulates a Newport ESP301
+    Same interface as the serial port
+    """
+
+    def __init__(self, timeout=1):
+        # we don't care about the actual parameters but timeout
+        self.timeout = timeout
+        self._output_buf = ""  # what the commands sends back to the "host computer"
+        self._input_buf = ""  # what we receive from the "host computer"
+
+        self._position = [0, 0, 0]
+        self._start_pos = [0, 0, 0]
+        self._range = [-500, 500]
+        self._target = [0, 0, 0]
+        self._speed = [50, 50, 50]
+        self._accel = [20, 20, 20]
+        self._decel = [20, 20, 20]
+        self._error_stack = []
+        
+        self._time_start = time.time()
+        self._current_move_finish = time.time()
+        self._moving = False
```

Comment:
Instead of having a `_moving` variable, which must be correctly updated (and depends on the user calling MD), you could have a simple function which looks like:
`return time.time() < self._current_move_finish`.
When stopping a move, just do `self._current_move_finish = 0`.